### PR TITLE
Update the answer in the text field while the user is typing

### LINKED ISSUE #1498: Answer change listener only called on focus shift to other text / button
**Describe the bug**
The new changes where answer change listener is called only when focus moves out of edit-text makes makes running calculations impossible for forms which do not have any other edit-text into form. 

**Component**
SDC library

**Expected behavior**
In screenshot below the age input is supposed to trigger onChange when value is updated. For edit-text it does not trigger until focus moves out of the widget. The only possibility to make that is possible is to add another edit-text and click into its box to run onChange for age-text. No click on anywhere on screen, on boolean widget, on opening date widget does that. Clicking on back button although triggers onChange for age. It should run onChange for age when clicking anywhere and on any other widget on screen.

**Screenshots and Video**

 GIF

**Would you like to work on the issue?**
Please state if this issue should be assigned to you or who you think could help to solve this issue. @jingtang10

### PR DESCRIPTION
**Pull request by jingtang10**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #1498 

**Description**
This PR updates the answer in the text field while the user is typing. Additionally, it does:
- fixes a NPE in the answer comparison in QuestionnaireItemViewItem and addes tests
- disables animation in recycler view which causes focus issues

**Alternative(s) considered**
NA

**Type**
Bug fix

**Screenshots (if applicable)**

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://github.com/google/android-fhir/wiki/Contributing) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

**Comment by jingtang10** [comment]
@maimoonak fyi

**Comment by maimoonak** [comment]
Looks like it would resolve the problem with onChange with edit-text. Just curious about the following
```
textInputEditText.setOnFocusChangeListener { view, focused ->
      if (!focused) {
        (view.context.applicationContext.getSystemService(Context.INPUT_METHOD_SERVICE) as
            InputMethodManager)
          .hideSoftInputFromWindow(view.windowToken, 0)
        val input = getValue(textInputEditText.editableText.toString())
        if (input != null) {
          questionnaireItemViewItem.setAnswer(input)
        } else {
          questionnaireItemViewItem.clearAnswer()
        }
      }
    }
```

Since now we would be notified for each text change this may trigger onChange twice for last input. Is it needed now? OR do we need setAnswer part of this?

**Comment by jingtang10** [comment]
> Looks like it would resolve the problem with onChange with edit-text. Just curious about the following
> 
> ```
> textInputEditText.setOnFocusChangeListener { view, focused ->
>       if (!focused) {
>         (view.context.applicationContext.getSystemService(Context.INPUT_METHOD_SERVICE) as
>             InputMethodManager)
>           .hideSoftInputFromWindow(view.windowToken, 0)
>         val input = getValue(textInputEditText.editableText.toString())
>         if (input != null) {
>           questionnaireItemViewItem.setAnswer(input)
>         } else {
>           questionnaireItemViewItem.clearAnswer()
>         }
>       }
>     }
> ```
> 
> Since now we would be notified for each text change this may trigger onChange twice for last input. Is it needed now? OR do we need setAnswer part of this?

actually just tested and i remember why i didn't take it out now. the user might focus on the text input and move away without typing anything. in this case we still want to run validation.

**Review by aditya-07 — APPROVED** [review]
Just a small comment around test case. Otherwise, LGTM!

File: datacapture/src/test/java/com/google/android/fhir/datacapture/views/QuestionnaireItemViewItemTest.kt [hunk 1]
```diff
@@ -213,4 +214,377 @@ class QuestionnaireItemViewItemTest {
...
+              validationResult = ValidationResult(true, listOf()),
+              answersChangedCallback = { _, _, _ -> }
+            )
+          )
+      )
+      .isTrue()
+  }
+
+  @Test
+  fun `hasTheSameValidationResult() should return false for different validation results`() {
```
[the remark this answers is not in the document]
Comment (jingtang10) [reply datacapture/src/test/java/com/google/android/fhir/datacapture/views/QuestionnaireItemViewItemTest.kt:550]: done

**Pull request merged**